File: warehouse-frontend/src/api/kanban.ts
import request from './request'

export interface Kanban {
  id: number
  kanbanNo: string
  inboundOrderId: number
  inboundOrderNo: string
  outboundOrderId?: number
  outboundOrderNo?: string
  outboundScanTime?: string
  partId: number
  partCode: string
  partName: string
  supplierName: string
  quantity: number
  boxSeq: number
  warehouseAreaId: number
  warehouseAreaName: string
  status: number
  statusText: string
  createTime: string
}

export function listKanbansByOrder(orderId: number): Promise<{
  code: number
  message: string
  data: Kanban[]
}> {
  return request.post('/kanban/list-by-order', { orderId })
}

export function listKanbansByPart(partId: number): Promise<{
  code: number
  message: string
  data: Kanban[]
}> {
  return request.post('/kanban/list-by-part', { partId })
}

/** 封存看板 */
export function blockKanbanApi(kanbanNo: string): Promise<{
  code: number
  message: string
  data: null
}> {
  return request.post('/kanban/block', { kanbanNo })
}

/** 解封看板 */
export function unblockKanbanApi(kanbanNo: string): Promise<{
  code: number
  message: string
  data: null
}> {
  return request.post('/kanban/unblock', { kanbanNo })
}

/** 按零件批量封存 */
export function blockByPartApi(partId: number): Promise<{
  code: number
  message: string
  data: number
}> {
  return request.post('/kanban/block-by-part', { partId })
}

/** 批量解封 */
export function batchUnblockApi(kanbanNos: string[]): Promise<{
  code: number
  message: string
  data: number
}> {
  return request.post('/kanban/batch-unblock', { kanbanNos })
}

/** 扫码翻转封存/解封 */
export function toggleBlockApi(kanbanNo: string): Promise<{
  code: number
  message: string
  data: {
    kanbanNo: string
    partName: string
    partCode: string
    action: string
    previousStatus: number
    previousStatusText: string
    newStatus: number
  }
}> {
  return request.post('/kanban/toggle-block', { kanbanNo })
}

/** 看板状态映射 */
export const KanbanStatusText: Record<number, string> = {
  0: '待入库',
  1: '在库可用',
  2: '待出库',
  3: '已出库',
  4: '封存',
}

export const KanbanStatusTagType: Record<number, string> = {
  0: 'info',
  1: 'success',
  2: 'warning',
  3: 'danger',
  4: 'info',
}

File: warehouse-backend/src/main/java/com/warehouse/model/entity/Kanban.java
package com.warehouse.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("kanban")
public class Kanban {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String kanbanNo;
    private Long inboundOrderId;
    private String inboundOrderNo;
    private Long outboundOrderId;
    private String outboundOrderNo;
    private Long partId;
    private String partCode;
    private String partName;
    private String supplierName;
    private Integer quantity;
    private Integer originalQty;   // 箱容量（原始装箱数量，不受转包影响）
    private Integer boxSeq;
    private Long warehouseAreaId;
    private String warehouseAreaName;
    private Integer status;
    private LocalDateTime createTime;

    // 状态常量
    public static final int STATUS_PENDING_INBOUND = 0;
    public static final int STATUS_AVAILABLE = 1;
    public static final int STATUS_LOCKED = 2;
    public static final int STATUS_OUTBOUND = 3;
    public static final int STATUS_BLOCKED = 4;
    public static final int STATUS_PARTIAL_REPACK = 5; // 部分转出（剩余>0，仍可被出库匹配）
    public static final int STATUS_CLEARED = 6;        // 已清空（剩余=0，完全转出或合并后作废）

    // 出库扫码时间（非表字段，查询时动态填充）
    @TableField(exist = false)
    private LocalDateTime outboundScanTime;

    public String getStatusText() {
        switch (this.status) {
            case 0: return "待入库";
            case 1: return "在库可用";
            case 2: return "待出库";
            case 3: return "已出库";
            case 4: return "封存";
            case 5: return "部分转出";
            case 6: return "已清空";
            default: return "未知";
        }
    }
}
